### ISSUE TITLE
/bin/Xorg has incorrect permissions

### ISSUE THREAD

**Issue by jens-maus** [Thinstation/thinstation#72]
Recently our Thinstation installation stopped working for certain thin clients. After investigation we found that these clients are using ATI Radeon cards and thus are loading the radeon driver for Xorg:

```
...
[   179.936] (II) RADEON(0): EXA: Driver will allow EXA pixmaps in VRAM
[   179.936] (==) RADEON(0): DPI set to (96, 96)
[   179.937] (II) Loading sub module "fb"
[   179.937] (II) LoadModule: "fb"
[   179.939] (II) Loading /usr/lib/xorg/modules/libfb.so
[   179.942] (II) Module fb: vendor="X.Org Foundation"
[   179.942] 	compiled for 1.17.1, module version = 1.0.0
[   179.943] 	ABI class: X.Org ANSI C Emulation, version 0.4
[   179.943] (II) Loading sub module "ramdac"
[   179.943] (II) LoadModule: "ramdac"
[   179.944] (II) Module "ramdac" already built-in
[   179.944] (II) UnloadModule: "modesetting"
[   179.945] (II) Unloading modesetting
[   179.945] (II) UnloadModule: "fbdev"
[   179.946] (II) Unloading fbdev
[   179.946] (II) UnloadSubModule: "fbdevhw"
[   179.947] (II) Unloading fbdevhw
[   179.948] (--) Depth 24 pixmap format is 32 bpp
[   179.948] Unable to retrieve master
[   179.949] (EE) 
Fatal server error:
[   179.950] (EE) AddScreen/ScreenInit failed for driver 0
[   179.950] (EE) 
[   179.951] (EE) 
Please consult the ThinStation made by Don Cupp support 
	 at http://www.thinstation.org
 for help. 
[   179.952] (EE) Please also check the log file at "/home/tsuser/.local/share/xorg/Xorg.0.log" for additional information.
[   179.953] (EE) 
[   180.024] (EE) Server terminated with error (1). Closing log file.
```

After investigation via 'strace' we found:

```
open("/dev/dri/card0", O_RDWR|O_LARGEFILE) = 11
ioctl(11, 0xc0106407, 0xff94daf0)       = 0
...
ioctl(11, 0xc018645c, 0xff94dc48)       = 0
ioctl(11, 0x641e, 0)                    = -1 EACCES (Permission denied)
```

After some searching on the web we found the following reference to the exact same problem:

http://unix.stackexchange.com/questions/178883/running-x-without-root

There people were having the same problem when starting Xorg in non-root mode. Thus, we further investigated and identified that compare to our old thinstation version the latest master branch of thinstation also starts Xorg as 'tsuser' and not as 'root. Thus, we investigated regarding the reasons for that:

Old Xorg file permission are:
```
# ls -la /bin/Xorg 
-rwsr-xr-x    1 root     root       2233228 May 20 11:37 /bin/Xorg
```

Latest master branch of thinstation shows:

```
# ll /bin/Xorg 
-rwxr-xr-x    1 tsuser   tsuser     2233228 Jun 10 13:56 /bin/Xorg
```

Thus, the suid bit and user binding to 'root' have been removed.

The following change does, however, restore this old permission setup and get Xorg working again for the thin clients with radeon cards:

https://github.com/hzdr/thinstation/commit/d5309c2511abd1f64c52a99d3c92afb9f96646bd

Would be great to identify the root of the issue appearing and to proper fix it in the upstream version of thinstation. Any idea what this might cause?

**Comment by Doncuppjr** [comment]
I think that both of these issues might be resolved by recent commits that I made for the HOME dir on the running client and LANG in the build environment. Please merge master and let me know if either issue persists. 


     On Wednesday, June 10, 2015 6:06 AM, Jens Maus <notifications@github.com> wrote:
   

 Recently our Thinstation installation stopped working for certain thin clients. After investigation we found that these clients are using ATI Radeon cards and thus are loading the radeon driver for Xorg:...
[   179.936] (II) RADEON(0): EXA: Driver will allow EXA pixmaps in VRAM
[   179.936] (==) RADEON(0): DPI set to (96, 96)
[   179.937] (II) Loading sub module "fb"
[   179.937] (II) LoadModule: "fb"
[   179.939] (II) Loading /usr/lib/xorg/modules/libfb.so
[   179.942] (II) Module fb: vendor="X.Org Foundation"
[   179.942]    compiled for 1.17.1, module version = 1.0.0
[   179.943]    ABI class: X.Org ANSI C Emulation, version 0.4
[   179.943] (II) Loading sub module "ramdac"
[   179.943] (II) LoadModule: "ramdac"
[   179.944] (II) Module "ramdac" already built-in
[   179.944] (II) UnloadModule: "modesetting"
[   179.945] (II) Unloading modesetting
[   179.945] (II) UnloadModule: "fbdev"
[   179.946] (II) Unloading fbdev
[   179.946] (II) UnloadSubModule: "fbdevhw"
[   179.947] (II) Unloading fbdevhw
[   179.948] (--) Depth 24 pixmap format is 32 bpp
[   179.948] Unable to retrieve master
[   179.949] (EE) 
Fatal server error:
[   179.950] (EE) AddScreen/ScreenInit failed for driver 0
[   179.950] (EE) 
[   179.951] (EE) 
Please consult the ThinStation made by Don Cupp support 
     at http://www.thinstation.org
 for help. 
[   179.952] (EE) Please also check the log file at "/home/tsuser/.local/share/xorg/Xorg.0.log" for additional information.
[   179.953] (EE) 
[   180.024] (EE) Server terminated with error (1). Closing log file.
After investigation via 'strace' we found:open("/dev/dri/card0", O_RDWR|O_LARGEFILE) = 11
ioctl(11, 0xc0106407, 0xff94daf0)       = 0
...
ioctl(11, 0xc018645c, 0xff94dc48)       = 0
ioctl(11, 0x641e, 0)                    = -1 EACCES (Permission denied)
After some searching on the web we found the following reference to the exact same problem:http://unix.stackexchange.com/questions/178883/running-x-without-rootThere people were having the same problem when starting Xorg in non-root mode. Thus, we further investigated and identified that compare to our old thinstation version the latest master branch of thinstation also starts Xorg as 'tsuser' and not as 'root. Thus, we investigated regarding the reasons for that:Old Xorg file permission are:# ls -la /bin/Xorg 
-rwsr-xr-x    1 root     root       2233228 May 20 11:37 /bin/Xorg
Latest master branch of thinstation shows:# ll /bin/Xorg 
-rwxr-xr-x    1 tsuser   tsuser     2233228 Jun 10 13:56 /bin/Xorg
Thus, the suid bit and user binding to 'root' have been removed.The following change does, however, restore this old permission setup and get Xorg working again for the thin clients with radeon cards:hzdr@d5309c2Would be great to identify the root of the issue appearing and to proper fix it in the upstream version of thinstation. Any idea what this might cause?—

**Comment by jens-maus** [comment]
Yes, both problems are gone now. However, there seems to be still issues with some of the init scripts part of packages. For example, within the "thinlinc" package (build/packages/thinlinc/etc/init.d/thinlinc.init) directories and files are generated. However, ${HOME} is used there, but ${HOME} is defined to / at the time these init scripts are running and thus everything ends up at the wrong place.

The same applied for the known_hosts file specified in the build.conf. Thinlinc is now creating a ".ssh" directory with a "known_hosts" file in / instead of /home/tsuser.

And fix for both things?

**Comment by Doncuppjr** [comment]
I think Thinlinc is the only third party package that directly integrates with thinstaiton, and as such, includes it's own init script. Incindio would have to update their script for a permanent fix to this issue. I will update network_files to fix the .ssh


     On Thursday, June 11, 2015 4:50 AM, Jens Maus <notifications@github.com> wrote:
   

 Yes, both problems are gone now. However, there seems to be still issues with some of the init scripts part of packages. For example, within the "thinlinc" package (build/packages/thinlinc/etc/init.d/thinlinc.init) directories and files are generated. However, ${HOME} is used there, but ${HOME} is defined to / at the time these init scripts are running and thus everything ends up at the wrong place.The same applied for the known_hosts file specified in the build.conf. Thinlinc is now creating a ".ssh" directory with a "known_hosts" file in / instead of /home/tsuser.And fix for both things?—

**Comment by jens-maus** [comment]
ok, thank you. .ssh is now fixed and I added a workaround for the now broken thinlinc init script. Someone should, however, notify cendio that they should update their package accordingly. Or we could do our own package. In fact, once I created a thinlinc_nightly package which downloads the thinlinc linux binaries only and not the thinlinc package and then extract everything. See here: https://github.com/hzdr/thinstation/tree/master/ts/build/packages/thinlinc_nightly
